modules/libcom/src/iocsh/initHooks.h

```diff
@@ -36,33 +45,57 @@ typedef enum {
+    initHookAfterIocBuilt,          /**< \brief End of iocBuild command */
 
-    initHookAtIocRun,               /* Start of iocRun command */
+    initHookAtIocRun,               /**< \brief Start of iocRun command */
     initHookAfterDatabaseRunning,
     initHookAfterCaServerRunning,
-    initHookAfterIocRunning,        /* End of iocRun/iocInit commands */
+    initHookAfterIocRunning,        /**< \brief End of iocRun/iocInit commands */
 
-    initHookAtIocPause,             /* Start of iocPause command */
+    initHookAtIocPause,             /**< \brief Start of iocPause command */
     initHookAfterCaServerPaused,
     initHookAfterDatabasePaused,
-    initHookAfterIocPaused,         /* End of iocPause command */
+    initHookAfterIocPaused,         /**< \brief End of iocPause command */
 
-    initHookAtShutdown,             /* Start of iocShutdown commands */
+    initHookAtShutdown,             /**< \brief Start of iocShutdown commands */
     initHookAfterCloseLinks,
-    initHookAfterStopScan,          /* triggered only by unittest code.   testIocShutdownOk() */
-    initHookAfterStopCallback,      /* triggered only by unittest code.   testIocShutdownOk() */
+    initHookAfterStopScan,          /**< \brief triggered only by unittest code.   testIocShutdownOk() */
+    initHookAfterStopCallback,      /**< \brief triggered only by unittest code.   testIocShutdownOk() */
     initHookAfterStopLinks,
-    initHookBeforeFree,             /* triggered only by unittest code.   testIocShutdownOk() */
-    initHookAfterShutdown,          /* End of iocShutdown commands */
+    initHookBeforeFree,             /**< \brief triggered only by unittest code.   testIocShutdownOk() */
+    initHookAfterShutdown,          /**< \brief End of iocShutdown commands */
+    initHookAfterInterruptAccept,   /**< \brief After initHookAfterDatabaseRunning */
+    initHookAtEnd,                  /**< \brief Before initHookAfterIocRunning */
+} initHookState;
 
-/* Deprecated states, provided for backwards compatibility.
- * These states are announced at the same point they were before,
- * but will not be repeated if the IOC gets paused and restarted.
+/** \code {.cpp}
+ * Any functions that are registered before iocInit reaches the desired state will be called when it reaches that state. 
+ * The initHookName function returns a static string representation of the state passed into it which is intended for printing. The following skeleton code shows how to use this facility:
+ *
+ * static initHookFunction myHookFunction; 
+ *
+ *int myHookInit(void) 
+ * { 
+ *  return(initHookRegister(myHookFunction)); 
+ *} 
+ * 
+ * static void myHookFunction(initHookState state) 
+ * { 
+ *  switch(state) { 
+ *    case initHookAfterInitRecSup: 
+ *      ... 
+ *      break; 
+ *   case initHookAfterInterruptAccept: 
+ *      ... 
+ *      break; 
+ *    default: 
+ *      break; 
+ *  } 
+ * } 
+ * An arbitrary number of functions can be registered. 
+ * \endcode 
```

Comment:
If done this way, the english paragraph would end up in the code snippet in the generated documentation. You can try this instead:

```suggestion
/**
 * \brief Any functions that are registered before iocInit reaches the desired state will be called when it reaches that state. 
 * The initHookName function returns a static string representation of the state passed into it which is intended for printing. The following skeleton code shows how to use this facility:
 *
 * \code{.cpp}
 * static initHookFunction myHookFunction; 
 *
 * int myHookInit(void) 
 * { 
 *   return(initHookRegister(myHookFunction)); 
 * } 
 * 
 * static void myHookFunction(initHookState state) 
 * { 
 *  switch(state) { 
 *    case initHookAfterInitRecSup: 
 *      ... 
 *      break; 
 *   case initHookAfterInterruptAccept: 
 *      ... 
 *      break; 
 *    default: 
 *      break; 
 *  } 
 * } 
 * \endcode 
 *
 * An arbitrary number of functions can be registered. 
```

Reply:
Also, the suggested format will not show new lines. This is why I added \n in my first version to make the example readable an d useful .

Reply:
I'm not sure I see the problem you are having. Could you share a screenshot?

I tried it on my computer and here is the result before:

![before](https://user-images.githubusercontent.com/1200507/168340273-7f88854d-fb9b-40e6-b8ab-9626d818fb70.png)

and after:

![after](https://user-images.githubusercontent.com/1200507/168340284-c1c7e832-6f2d-44c3-9502-0cfea2122786.png)